app/pluginmanager.js

```diff
@@ -1142,6 +1142,8 @@ PluginManager.prototype.checkPluginDependencies = function (folder) {
   
   self.logger.info('Check plugin dependencies');
 
+  // Check for native addons
+  var native_modules = execSync(`find ${folder}/node_modules -name obj.target -prune -false -o -type f -name "*.node" 2>/dev/null`,{ encoding: 'utf8' }).split('\n').filter(n=>n);
```

Comment:
Two things:
- Where possible, we should avoid sync operations, since they slow system quite a lot. In this case however it does not really matter since this is done on installing the plugin.
- When using execsync, we shall always wrap in try catch, otherwise if there is an error the system will crash.
- We need also to check existing plugins, not the newly installed ones.

Reply:
> Where possible, we should avoid sync operations

This is doubly true now that we can use native promises and async/await.

Reply:
>When using execsync, we shall always wrap in try catch, otherwise if there is an error the system will crash.

```<> 2>/dev/null``` should not lead to a trapable error, but will do. 

>We need also to check existing plugins, not the newly installed ones.

Existing plugins will already disable themselves, and throw an error notification. 
(how useful that is is debatable, as the webUI won't be up during first plugin startup)